app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt

```diff
@@ -0,0 +1,137 @@
+package com.jetpack.compose.learning.magnifierview
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.MotionEvent
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInteropFilter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import com.jetpack.compose.learning.R
+import com.jetpack.compose.learning.theme.AppThemeState
+import com.jetpack.compose.learning.theme.BaseView
+import com.jetpack.compose.learning.theme.SystemUiController
+
+
+@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
+class MagnifierViewActivity : AppCompatActivity() {
+
+    companion object {
+        const val MAGNIFIER_ZOOM_FACTOR = 2F
+    }
+    private val Offset.Companion.default
+        get() = Offset(0F, 1000F)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            MainContent()
+        }
+    }
+
+    @Composable
+    private fun MainContent() {
+        val systemUiController = remember { SystemUiController(window) }
+        val appTheme = remember { mutableStateOf(AppThemeState()) }
+        BaseView(appTheme.value, systemUiController) {
+            Scaffold(
+                topBar = {
+                    TopAppBar(title = { Text("MagnifierView") }, navigationIcon = {
+                        IconButton(onClick = { onBackPressed() }) {
+                            Icon(Icons.Filled.ArrowBack, contentDescription = null)
+                        }
+                    })
+                }
+            ) {
+                MagnifierScreen()
+            }
+        }
+    }
+
+
+
+    @OptIn(ExperimentalFoundationApi::class, ExperimentalComposeUiApi::class)
+    @Composable
+    fun MagnifierScreen() {
```

Comment:
Kindly rename this function to "MagnifierViewComposable". It's been a good practice to use "Composable" postfix to any Composable functions.